### PR TITLE
better error handling in the runtime

### PR DESCRIPTION
**Pull request by dmah42**
using thiserror more properly enables the runtime mod.rs
functionality to have less noise.

### CODE REVIEW

File: auraed/src/runtime/error.rs [hunk 1]
```diff
@@ -2,27 +2,40 @@ use log::error;
...
 use tonic::Status;
 
-#[derive(Error, Debug, PartialEq, Eq)]
-pub enum CellServiceError {
+pub(crate) type Result<T> = std::result::Result<T, RuntimeError>;
+
+#[derive(Error, Debug)]
+pub(crate) enum RuntimeError {
     #[error("missing argument in request: {arg}")]
     MissingArgument { arg: String },
```
Comment (future-highway) [inline auraed/src/runtime/error.rs:10]: We can take this out as validation takes care of it.

File: auraed/src/runtime/error.rs [hunk 2]
```diff
@@ -2,27 +2,40 @@ use log::error;
...
+    ValidationError(#[from] validation::ValidationError),
+
     #[error("internal error: {msg}: {err}")]
     Internal { msg: String, err: String },
+
+    #[error(transparent)]
+    CgroupsError(#[from] cgroups_rs::error::Error),
+
+    #[error(transparent)]
+    Other(#[from] anyhow::Error),
```
Comment (future-highway) [inline auraed/src/runtime/error.rs:22]: Ideally we will eventually eliminate anyhow (I think)

File: auraed/src/runtime/error.rs [hunk 3]
```diff
@@ -2,27 +2,40 @@ use log::error;
...
 }
 
-impl From<CellServiceError> for Status {
-    fn from(err: CellServiceError) -> Self {
+impl From<RuntimeError> for Status {
```
Comment (future-highway) [inline auraed/src/runtime/error.rs:25]: Looks like you were a step ahead of me, lol

File: auraed/src/runtime/error.rs (hunk 1 above)
Comment (dmah42) [reply auraed/src/runtime/error.rs:10]: i tried, but i needed this to go from ValidationError to RuntimeError to Status.

i didn't see anything that would convert ValidationError straight to Status.. did i miss it?

File: auraed/src/runtime/error.rs (hunk 2 above)
Comment (dmah42) [reply auraed/src/runtime/error.rs:22]: i agree.. though it's convenient for creating these quick errors from a string.  i think if we stick to `anyhow!` _only_ we might be ok.

File: auraed/src/runtime/error.rs (hunk 3 above)
Comment (dmah42) [reply auraed/src/runtime/error.rs:28]: :D

**Review by future-highway — DISMISSED** [review]

**Review by future-highway — APPROVED** [review]

**Pull request merged**